### PR TITLE
curlpp: new package at 0.8.1

### PR DESCRIPTION
**Pull request by djt3**
## Description of package
A c++ curl wrapper

## Installed manifest of package

(`cat /var/db/kiss/installed/<pkg_name>/manifest`)

```
/var/db/kiss/installed/curlpp/version
/var/db/kiss/installed/curlpp/sources
/var/db/kiss/installed/curlpp/manifest
/var/db/kiss/installed/curlpp/depends
/var/db/kiss/installed/curlpp/checksums
/var/db/kiss/installed/curlpp/build
/var/db/kiss/installed/curlpp/
/var/db/kiss/installed/
/var/db/kiss/
/var/db/
/var/
/usr/lib/pkgconfig/curlpp.pc
/usr/lib/pkgconfig/
/usr/lib/libcurlpp.so.1.0.0
/usr/lib/libcurlpp.so.1
/usr/lib/libcurlpp.so
/usr/lib/libcurlpp.a
/usr/lib/
/usr/include/utilspp/clone_ptr.hpp
/usr/include/utilspp/TypeTrait.hpp
/usr/include/utilspp/TypeList.hpp
/usr/include/utilspp/ThreadingSingle.inl
/usr/include/utilspp/ThreadingSingle.hpp
/usr/include/utilspp/ThreadingFactoryMutex.inl
/usr/include/utilspp/ThreadingFactoryMutex.hpp
/usr/include/utilspp/SmartPtr.hpp
/usr/include/utilspp/NullType.hpp
/usr/include/utilspp/NonCopyable.hpp
/usr/include/utilspp/EmptyType.hpp
/usr/include/utilspp/
/usr/include/curlpp/internal/SList.hpp
/usr/include/curlpp/internal/OptionSetter.inl
/usr/include/curlpp/internal/OptionSetter.hpp
/usr/include/curlpp/internal/OptionList.hpp
/usr/include/curlpp/internal/OptionContainerType.hpp
/usr/include/curlpp/internal/OptionContainer.inl
/usr/include/curlpp/internal/OptionContainer.hpp
/usr/include/curlpp/internal/CurlHandle.inl
/usr/include/curlpp/internal/CurlHandle.hpp
/usr/include/curlpp/internal/
/usr/include/curlpp/cURLpp.hpp
/usr/include/curlpp/Types.hpp
/usr/include/curlpp/Options.hpp
/usr/include/curlpp/OptionBase.hpp
/usr/include/curlpp/Option.inl
/usr/include/curlpp/Option.hpp
/usr/include/curlpp/Multi.hpp
/usr/include/curlpp/Infos.hpp
/usr/include/curlpp/Info.inl
/usr/include/curlpp/Info.hpp
/usr/include/curlpp/Form.hpp
/usr/include/curlpp/Exception.hpp
/usr/include/curlpp/Easy.inl
/usr/include/curlpp/Easy.hpp
/usr/include/curlpp/
/usr/include/
/usr/bin/curlpp-config
/usr/bin/
/usr/

```

## New package

- [Y] Latest upstream version.
- [Y] I agree to maintain this package.
- [Y] I agree to notify @dylanaraps if I can no longer maintain this package.

### CODE REVIEW

File: community/curlpp/build [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh -e
+
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX='/usr' -DCMAKE_INSTALL_LIBDIR='/usr/lib' ../
+make
+make DESTDIR="$1" install
```
Comment (dylanaraps) [inline community/curlpp/build:7]: This isn't how we structure cmake based builds.

See: https://k1ss.org/style (0800)

```sh
export DESTDIR="$1"                                 
                                                   
cmake -B build \                                    
   -DCMAKE_INSTALL_PREFIX=/usr \                   
   -DFLAG=1                                        
                                                   
cmake --build   build                               
cmake --install build
```

The benefit of this syntax is that the user has the choice between Make or Ninja
for the build itself (configurable via CMAKE_GENERATOR). This is a feature of
cmake itself and nothing special to KISS.

Comment (djt3) [reply community/curlpp/build]: Very sorry about that, got a bit confused there. I have updated my fork with what I think is the correct syntax.

File: community/curlpp/build [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
-mkdir build
-cd build
-cmake -DCMAKE_INSTALL_PREFIX='/usr' -DCMAKE_INSTALL_LIBDIR='/usr/lib' ../
-make
-make DESTDIR="$1" install
+export DESTDIR="$1"
+
+cmake -B build \
+	-DCMAKE_INSTALL_PREFIX='/usr' \
+	-DCMAKE_INSTALL_LIBDIR='/usr/lib'
```
Comment (dylanaraps) [inline community/curlpp/build:7]: Indentation is four spaces. :+1:

Comment (djt3) [reply community/curlpp/build]: fixed it, thanks 👍

**Pull request merged**